### PR TITLE
[RISCV][GISel] Simplify selectSelect. NFC

### PR DESCRIPTION
**Pull request by topperc**
Use GSelect and reduce number of temporaries.

### CODE REVIEW

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 1]
```diff
@@ -776,27 +776,25 @@ static void getICMPOperandsForBranch(MachineInstr &MI, MachineIRBuilder &MIB,
...
-  Register MIOp1Reg = MI.getOperand(1).getReg();
-  bool Op1IsICMP = mi_match(MIOp1Reg, MRI, m_GICmp(m_Pred(), m_Reg(), m_Reg()));
-  RISCVCC::CondCode CC;
-  Register LHS, RHS;
+  Register LHS = SelectMI.getCondReg();
+  Register RHS = RISCV::X0;
+  RISCVCC::CondCode CC = RISCVCC::COND_NE;
+
+  bool Op1IsICMP = mi_match(LHS, MRI, m_GICmp(m_Pred(), m_Reg(), m_Reg()));
   if (Op1IsICMP)
```
Comment (mshockwave) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:788]: I think Op1IsICMP can be merged into the if statement (without making the line too long)

**Review by mshockwave — APPROVED** [review]
LGTM w/ a minor comment

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 2]
```diff
@@ -776,27 +776,24 @@ static void getICMPOperandsForBranch(MachineInstr &MI, MachineIRBuilder &MIB,
...
+    getICMPOperandsForBranch(*MRI.getVRegDef(LHS), MIB, MRI, CC, LHS, RHS);
+
   MachineInstr *Result = MIB.buildInstr(RISCV::Select_GPR_Using_CC_GPR)
-                             .addDef(MI.getOperand(0).getReg())
-                             .addReg(Op1)
-                             .addReg(Op2)
-                             .addImm(Op3)
-                             .addReg(MI.getOperand(2).getReg())
-                             .addReg(MI.getOperand(3).getReg());
+                             .addDef(SelectMI.getReg(0))
```
Comment (michaelmaitland) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:791]: Do you think we can have another patch which adds `getDstReg` to GSelect? It looks like many other instructions have this function.

Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:791]: Got it.

**Review by michaelmaitland — APPROVED** [review]
LGTM.
